libs/eavmlib/src/ahttp_client.erl

```diff
@@ -0,0 +1,496 @@
+    ),
+    {UpdatedParser, Parsed} = maybe_handle_end_of_response(
+        ReplacedAccParser#parser_state{remaining_body_bytes = NewRemBodyBytes}, [
+            {data, Chunk} | ParsedAcc
+        ]
+    ),
+    {ok, UpdatedParser, Parsed}.
+
+consume_lines(Parser, ParsedAcc) ->
+    case binary:split(Parser#parser_state.acc, <<"\r\n">>) of
+        [_NotTerminatedLine] ->
+            {ok, Parser, ParsedAcc};
+        [Line, Rest] ->
+            ReplacedAccParser = replace_chunk(Parser, Rest),
+            case parse_line(ReplacedAccParser, Line) of
+                {consume_bytes, UpdatedParser} ->
+                    consume_bytes(UpdatedParser, ParsedAcc);
+                {ok, UpdatedParser} ->
+                    consume_lines(UpdatedParser, ParsedAcc);
+                {ok, UpdatedParser, Found} ->
+                    consume_lines(UpdatedParser, [Found | ParsedAcc]);
+                {error, UpdatedParser, NotParsed} ->
+                    {error, UpdatedParser, ParsedAcc, NotParsed}
+            end
+    end.
+
+append_chunk(#parser_state{acc = undefined} = Parser, Chunk) ->
+    Parser#parser_state{acc = Chunk};
+append_chunk(#parser_state{acc = Acc} = Parser, Chunk) ->
+    Parser#parser_state{acc = <<Acc/binary, Chunk/binary>>}.
+
+replace_chunk(Parser, <<>>) ->
+    Parser#parser_state{acc = undefined};
+replace_chunk(Parser, Chunk) ->
+    Parser#parser_state{acc = Chunk}.
+
+parse_line(
+    #parser_state{state = undefined} = Parser, <<"HTTP/1.0 ", C:3/binary, " ", _Txt/binary>>
+) ->
+    StatusCode = binary_to_integer(C),
+    {ok, Parser#parser_state{state = headers}, {status, StatusCode}};
+parse_line(
+    #parser_state{state = undefined} = Parser, <<"HTTP/1.1 ", C:3/binary, " ", _Txt/binary>>
+) ->
+    StatusCode = binary_to_integer(C),
+    {ok, Parser#parser_state{state = headers}, {status, StatusCode}};
+parse_line(#parser_state{state = headers} = Parser, <<>>) ->
+    {consume_bytes, Parser#parser_state{state = body}};
+parse_line(
+    #parser_state{state = headers, last_header = ignore} = Parser, <<C, _MultiLine/binary>>
+) when C == $\s orelse C == $\t ->
+    {ok, Parser};
+parse_line(
+    #parser_state{state = headers, last_header = LastH} = Parser, <<C, MultiLine/binary>>
+) when is_binary(LastH) andalso (C == $\s orelse C == $\t) ->
+    LTrimmedValue = trim_left_spaces(MultiLine, 0),
+    TrimmedValue = trim_right_spaces(LTrimmedValue, byte_size(LTrimmedValue)),
+    {ok, Parser, {header_continuation, {LastH, TrimmedValue}}};
+parse_line(#parser_state{state = headers, wanted_headers = WantedHeaders} = Parser, HeaderLine) ->
+    case split_header(WantedHeaders, HeaderLine) of
+        {ok, Name, Value} ->
+            LTrimmedValue = trim_left_spaces(Value, 0),
+            TrimmedValue = trim_right_spaces(LTrimmedValue, byte_size(LTrimmedValue)),
+            UpdatedParser =
+                case Name of
+                    <<"Content-Length">> ->
```

Comment:
`"Content-Length"` is from the server right? 

and thus parsing should be case insensitive?

Reply:
split_header was taking it from a list, so it was guaranteed to have some casing. I clarified it.